packages/composites/augur/src/dataProviders/sportsdataio.ts

```diff
@@ -64,45 +82,131 @@ interface CommonScores {
+): Promise<string> => {
+  if (sport !== 'nfl') return 'NO_INFO'
+
   const input = {
     id,
     data: {
       sport,
-      endpoint: 'schedule',
-      season,
+      endpoint: 'current-season',
     },
   }
 
   const response = await exec(input, context)
-  const filtered = (response.result as { GlobalGameID: number }[])
-    .filter(event => event.GlobalGameID != 0)
+  return response.data.result
+}
+
+const getTeams = async (
+  id: string,
+  sport: string,
+  exec: Execute,
+  context: AdapterContext,
+): Promise<NFLTeam[]> => {
+  if (sport !== 'nfl') return []
+
+  const input = {
+    id,
+    data: {
+      sport,
+      endpoint: 'teams',
+    },
+  }
+  const response = await exec(input, context)
+  return response.data.result
+}
 
+const getSchedule = async (
+  id: string,
+  sport: string,
+  exec: Execute,
+  context: AdapterContext,
+): Promise<TeamSchedule[]> => {
   switch (sport) {
     case 'nfl': {
-      return (filtered as NFLEvent[]).map(event => ({
+      let events: NFLEvent[] = []
+      const currentSeason = await getCurrentSeason(id, sport, exec, context)
+
+      for (const seasonPostfixKey of ['PRE', '', 'POST', 'STAR']) {
+        const input = {
+          id,
+          data: {
+            sport,
+            endpoint: 'schedule',
+            season: `${currentSeason}${seasonPostfixKey}`,
+          },
+        }
+        const response = await exec(input, context)
+        const filtered = (response.result as NFLEvent[]).filter((event) => event.GlobalGameID != 0)
+
+        events = [...events, ...filtered]
+      }
+
+      // Need full team names. API just returns an abbreviations
+      const teamNames = await getTeams(id, sport, exec, context)
+      events = events.map((event) => {
+        const homeTeam = teamNames.find(
+          ({ GlobalTeamID }) => GlobalTeamID === event.GlobalHomeTeamID,
+        )
+        const awayTeam = teamNames.find(
+          ({ GlobalTeamID }) => GlobalTeamID === event.GlobalAwayTeamID,
+        )
+
+        if (!homeTeam || !awayTeam) return event
+
+        return {
+          ...event,
+          AwayTeam: awayTeam.FullName,
+          HomeTeam: homeTeam.FullName,
+        }
+      })
+
+      return (events as NFLEvent[]).map((event) => ({
         Date: event.Date || event.Day,
         GameID: event.GlobalGameID,
+        AwayTeamName: event.AwayTeam,
         AwayTeamID: event.GlobalAwayTeamID,
+        HomeTeamName: event.HomeTeam,
         HomeTeamID: event.GlobalHomeTeamID,
         Status: event.Status,
         PointSpread: event.PointSpread,
+        AwayTeamMoneyLine: event.AwayTeamMoneyLine,
+        HomeTeamMoneyLine: event.HomeTeamMoneyLine,
+        OverUnder: event.OverUnder,
       }))
     }
     case 'ncaa-fb': {
-      return (filtered as CFBGames[]).map(event => ({
+      const input = {
+        id,
+        data: {
+          sport,
+          endpoint: 'schedule',
+          season: '2021', // TODO: Find a source for this information.
```

Comment:
Hard-coding a year seems like a bad idea, as it's easily forgotten. Can we come up with a solution for this before merging?